### PR TITLE
Choice prompt null locale fix [Cherry-pick against 4.6]

### PR DESCRIPTION
**Pull request by mdrichardson**
Per standup, this is a cherry pick against 4.6 of [this already-merged-with-master PR](https://github.com/microsoft/botbuilder-js/pull/1400)

---

Parity with: https://github.com/microsoft/botbuilder-dotnet/pull/2943

# Changes

* `ChoicePrompt`: Ensures that locale defaults to English and handles mapping to the nearest locale more gracefully
* `PromptCultureModels`: Implements our own version of [Recognizers-Text's `MapToNearestLanguage`](https://github.com/microsoft/Recognizers-Text/blob/7b20c1bbbfdebc3a801fbc61b094706ceebd7acf/.NET/Microsoft.Recognizers.Text/Culture.cs#L66) -- They added a bunch of other supported languages that will break this without doing this.
* `ChoicePromptTests`: Added a test to specifically set the Locale to null

# Other Notes

* ConfirmPrompt already handled this fine and had tests for it
* I'll also work on some tests for PromptCultureModels, but I figured this should get out the door, first.
* Since Recognizers-Text added a bunch of supported languages, I'll see about adding them here in a future PR

## Workarounds until this is Merged

* Downgrade to < 4.6, or
* set `defaultLocale` on `ChoicePrompt`

### CODE REVIEW

File: libraries/botbuilder-dialogs/src/prompts/promptCultureModels.ts [hunk 1]
```diff
@@ -115,12 +115,35 @@ export class PromptCultureModels {
...
-    public static mapToNearestLanguage = (cultureCode: string): string => Culture.mapToNearestLanguage(cultureCode);
+    public static mapToNearestLanguage(cultureCode: string): string {
+        if (cultureCode !== undefined) {
+            cultureCode = cultureCode.toLowerCase();
+            let supportedCultureCodes = this.getSupportedCultureCodes();
+    
+            if (supportedCultureCodes.indexOf(cultureCode) < 0) {
+                let culturePrefix = cultureCode.split('-')[0].trim();
+    
+                supportedCultureCodes.forEach(function(supportedCultureCode): void {
```
Comment (christopheranderson) [inline libraries/botbuilder-dialogs/src/prompts/promptCultureModels.ts]: prefer anonymous functions

File: libraries/botbuilder-dialogs/src/prompts/promptCultureModels.ts [hunk 2]
```diff
@@ -115,12 +115,35 @@ export class PromptCultureModels {
...
+
     /**
      * Use Recognizers-Text to normalize various potential Locale strings to a standard.
+     * @remarks This is mostly a copy/paste from https://github.com/microsoft/Recognizers-Text/blob/master/JavaScript/packages/recognizers-text/src/culture.ts#L39
+     *          This doesn't directly use Recognizers-Text's MapToNearestLanguage because if they add language support before we do, it will break our prompts.
      * @param cultureCode Represents locale. Examples: "en-US, en-us, EN".
      * @returns Normalized locale.
      */
-    public static mapToNearestLanguage = (cultureCode: string): string => Culture.mapToNearestLanguage(cultureCode);
+    public static mapToNearestLanguage(cultureCode: string): string {
```
[the remark this answers is not in the document]
Comment (mdrichardson) [reply libraries/botbuilder-dialogs/src/prompts/promptCultureModels.ts]: If used directly, yes, it's optional and returns `undefined`. For uses in our SDK currently, it's always passed in *something*, defaulting to "en-us". I could see making it a required argument, though.

**Comment by mdrichardson** [comment]
@christopheranderson Don't commit any changes, then? it would be a pretty quick fix although I don't know the impact to the rest of the devops pipeline.

**Comment by christopheranderson** [comment]
It's best practice not to modify a cherry-pick unless it is required for branch compatibility, imo, so my recommendation is to ignore me. :)

**Pull request merged**